File: StrategyPattern/Bank/JAVA/SavingAccount.java
package ACCOUNT;

public class SavingAccount extends Account{

	boolean isExpired;
	
	public SavingAccount(boolean isExpiredCheck)
	{
		isExpired = isExpiredCheck;
		transaction = new SavingWithdrawal(isExpired, balance);
	}
	
/*	public void setExpired(boolean isExpiredCheck )
	{
		isExpired = isExpiredCheck;
		((SavingWithdrawal) transaction).setIsExpired(isExpiredCheck);
		System.out.println("Your Saving Account is expired");
	}
	*/
	@Override
	void display() {
		System.out.println("This is Saving Account");
		System.out.println("Balance  : " + balance);
		System.out.println("Account openning : " + openningDate.toString());
		System.out.println("Is account expired ? " + isExpired);
	}

}

File: StrategyPattern/Bank/C++/SavingAccount.cpp
#include "SavingAccount.h"


void SavingAccount::display()
{
	cout << "This is Minus Account " <<endl;
	cout << "Balance : " << balance <<endl;
	cout << "Account openning : " << openningDate.tm_year+ 1900 << "/" <<openningDate.tm_mon <<"/" <<
		openningDate.tm_mday <<" " << openningDate.tm_hour << ":" << openningDate.tm_min <<":"<< openningDate.tm_sec<< endl;
	cout << "Is account expired ?  ";
	if(isExpired)
		cout  << "yes"  <<endl;
	else
		cout  << "no"  <<endl;
}